src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -54,24 +72,38 @@ private Job runJob(Job queryJob) {
     }
   }
 
-  public List<String> runQuery(Query query) {
-    String queryString = generateQueryClause(query);
+  private Integer defaultValue(Integer val, Integer def) {
+    if (val != null) {
+      return val;
+    } else {
+      return def;
+    }
+  }
+
+  public QueryResult runQuery(Query query, Integer offset, Integer limit) {
+
+
+    String queryString = (new QueryTranslator(this.table, query)).sql();
+
     // Wrap query so it returns JSON
-    String jsonQuery = String.format("SELECT TO_JSON_STRING(t,true) from (%s) as t", queryString);
+    offset = defaultValue(offset, 0);
+    limit = defaultValue(limit, 100);
```

Comment:
This can be done with `Objects.requireNonNullElse()`
```suggestion
    offset = Objects.requireNonNullElse(offset, 0);
    limit = Objects.requireNonNullElse(limit, 100);
```